src/axom/inlet/LuaReader.hpp

```diff
@@ -92,16 +94,23 @@ class LuaReader : public Reader
    */
   static const int baseIndex = 1;
 
+protected:
   /*!
    *****************************************************************************
    * \brief Returns the Sol Lua state
    *
    * This allows the user to access functionality that was not provided by Inlet.
+   * 
+   * \note This is an advanced feature and could change the input file state after
+   *   it has been validated by Inlet. If you need to modify the Sol `state`,
+   *   be sure to include `axom/sol.hpp` and derive a new class to access this
+   *   publically. See the `lua_library.cpp` example. Including `sol.hpp` can
+   *   increase compile time significantly.
    *
-   * \return Reference to the Sol Lua state
+   * \return Shared pointer to the Sol Lua state
    *****************************************************************************
    */
-  axom::sol::state& solState() { return m_lua; }
+  std::shared_ptr<axom::sol::state> solState() { return m_lua; }
```

Comment:
Is there any reason/use case for the Sol state lifetime to extend beyond the lifetime of `LuaReader`? If not, I'd suggest making the `m_lua` member field an `std::unique_ptr` and just returning a reference to the state as before.

Reply:
I could see someone wanting the sol state to extend past when the LuaReader is done. The person that wanted this before was using quite a few things down the road after the input parsing was done.